src/commands/initialize-broadcast.js

```diff
@@ -1,42 +1,48 @@
 import State from '../state'
-import { logger } from '../helpers/constants'
+// import { logger } from '../helpers/constants'
 
 class InitializeBroadcast {
-  constructor(broadcastService) {
+  constructor({ broadcastService, messageService }) {
     this.broadcastService = broadcastService
+    this.messageService = messageService
     this.commandString = '/broadcast'
   }
 
-  shouldExecute(messageService) {
-    if (messageService.getCommand() === this.commandString) {
+  shouldExecute() {
+    if (this.messageService.getCommand() === this.commandString) {
       return true
     }
     return false
   }
 
-  execute(messageService) {
-    this.broadcastService.setMessage(messageService.getArgument())
-    this.broadcastService.setUserEmail(messageService.getUserEmail())
-    this.broadcastService.setVGroupID(messageService.getVGroupID())
+  execute() {
+    const {
+      argument,
+      // message,
+      userEmail,
+      vGroupID,
+    } = this.messageService.getMessageData()
+    console.log({ vGroupID })
+    this.broadcastService.setMessage(argument)
+    this.broadcastService.setUserEmail(userEmail)
+    this.broadcastService.setVGroupID(vGroupID)
     this.broadcastService.setTTL('')
     this.broadcastService.setBOR('')
     this.broadcastService.setSentByFlag(true)
-    logger.debug(`Here are all the things: ${messageService.getArgument()}`)
+    this.messageService.setUserCurrentState({ currentState: State.ASK_FOR_ACK }) // change to broacastservice?
```

Comment:
Why is currentState passed as an object?

Reply:
making the function args easier to understand and destructure, makes it easy to pinpoint errors. Most of our functions are now expecting objects. Typescript lite without typescript